optuna/visualization/matplotlib/_param_importances.py

```diff
@@ -90,37 +94,66 @@ def objective(trial):
 
     _imports.check()
 
-    importances_info = _get_importances_info(study, evaluator, params, target, target_name)
-    return _get_importances_plot(importances_info)
+    if target or not study._is_multi_objective():
+        importances_infos: tuple[_ImportancesInfo, ...] = (
+            _get_importances_info(study, evaluator, params, target, target_name),
+        )
 
+    else:
+        n_objectives = len(study.directions)
+        importances_infos = tuple(
+            _get_importances_info(
+                study,
+                evaluator,
+                params,
+                target=lambda t: t.values[objective_id],
+                target_name=f"{target_name} {objective_id}",
+            )
+            for objective_id in range(n_objectives)
+        )
 
-def _get_importances_plot(info: _ImportancesInfo) -> "Axes":
+    return _get_importances_plot(importances_infos)
+
+
+def _get_importances_plot(infos: tuple[_ImportancesInfo, ...]) -> "Axes":
     # Set up the graph style.
     plt.style.use("ggplot")  # Use ggplot style sheet for similar outputs to plotly.
     fig, ax = plt.subplots()
-    ax.set_title("Hyperparameter Importances")
-    ax.set_xlabel(f"Importance for {info.target_name}")
+    ax.set_title("Hyperparameter Importances", loc="left")
+    ax.set_xlabel("Hyperparameter Importance")
     ax.set_ylabel("Hyperparameter")
+    height = 0.8 / len(infos)  # Default height split between objectives.
+
+    for objective_id, info in enumerate(infos):
+        param_names = info.param_names
+        pos = np.arange(len(param_names))
+        offset = height * objective_id
+        importance_values = info.importance_values
+
+        if not importance_values:
+            continue
+
+        # Draw horizontal bars.
+        ax.barh(
+            pos + offset,
+            importance_values,
+            height=height,
+            align="center",
+            label=info.target_name,
+            color=plt.get_cmap("tab20c")(objective_id),
+        )
+
+        _set_bar_labels(info, fig, ax, offset)
+        ax.set_yticks(pos + offset / 2, param_names)
+
+    ax.legend(loc="best")
```

Comment:
Originally wanted to use [`outside right upper` loc](https://matplotlib.org/stable/tutorials/intermediate/legend_guide.html#figure-legends) which would make the legend position static and guarantee minimal overlap.
![Screenshot from 2023-08-13 12-19-00](https://github.com/optuna/optuna/assets/37713008/190fe921-d416-434a-b489-0d3c806bf168)


This however is not supported in Python 3.7, so going with loc `best` which still ensures minimal overlap with bars, but makes the legend position dynamic.